### PR TITLE
Store logs before anaconda starts (#1255659)

### PR DESCRIPTION
**Pull request by japokorn**
generates lsblk, dmesg a lvmdump reports before anaconda starts

Resolves: rhbz#1255659

### CODE REVIEW

File: data/systemd/anaconda-pre.service [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Wants=systemd-udev-settle.service
+Wants=NetworkManager.service
+Wants=plymouth-quit.service plymouth-quit-wait.service
+Wants=anaconda-direct.service anaconda.service
+Wants=anaconda-sshd.service
+Wants=zram.service
+Wants=systemd-logind.service
+
+[Service]
+Environment=HOME=/root PATH=/usr/bin:/bin:/sbin:/usr
```
Comment (jkonecny12) [inline data/systemd/anaconda-pre.service]: Do you need those Environment variables? I think you can use default here.

**Comment by jkonecny12** [comment]
Please rename the ``anaconda-pre-log-gen.sh`` script to ``anaconda-pre-log-gen``. Also this is causing your problems in Jenkins now.

File: data/systemd/anaconda-pre.service [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+Type=oneshot
+WorkingDirectory=/root
```
Comment (jkonecny12) [inline data/systemd/anaconda-pre.service]: The same as environment variables, I think you can remove this line.

File: data/systemd/anaconda-pre.service [hunk 3]
```diff
@@ -0,0 +1,25 @@
+[Unit]
+# This service is to be run before anaconda starts and log data before anaconda changes them
+Description=pre-anaconda logging service
```
Comment (jkonecny12) [inline data/systemd/anaconda-pre.service]: Better description would be nice. More what is this service doing. This is nice to have.

File: scripts/anaconda-pre-log-gen.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+# information about previous system state
+
+TARGET_DIRECTORY=/tmp/pre-anaconda-logs
+
+mkdir ${TARGET_DIRECTORY}
+
+lsblk -a >${TARGET_DIRECTORY}/block_devices.log
+dmesg -a >${TARGET_DIRECTORY}/block_devices.log
+
+lvmdump
```
Comment (jkonecny12) [inline scripts/anaconda-pre-log-gen.sh]: Use option -d with the directory. With this setup you will create ``lvmdump`` to /root.

**Comment by M4rtinK** [comment]
If you want the new logs to be copied over to the installed system then they need to be included in the built-in log copying post-script:
https://github.com/rhinstaller/anaconda/blob/rhel7-branch/data/post-scripts/99-copy-logs.ks#L11

File: scripts/anaconda-pre-log-gen.sh (hunk 4 above)
Comment (dwlehman) [inline scripts/anaconda-pre-log-gen.sh]: If you're going to collect `lvmdump` you should also get the contents of `/proc/mdstat` and the output of `dmsetup list` or similar.

File: data/systemd/anaconda-pre.service (hunk 1 above)
Comment (japokorn) [inline data/systemd/anaconda-pre.service]: Removed. Fixed.

File: data/systemd/anaconda-pre.service (hunk 2 above)
Comment (japokorn) [inline data/systemd/anaconda-pre.service]: `anaconda-pre-log-gen.sh` script renamed to `anaconda-pre-log-gen`

Comment (japokorn) [inline data/systemd/anaconda-pre.service]: environment variables removed. Fixed.

File: scripts/Makefile.am [hunk 5]
```diff
@@ -18,10 +18,10 @@
 # Author: David Cantrell <dcantrell@redhat.com>
 
 scriptsdir = $(libexecdir)/$(PACKAGE_NAME)
-dist_scripts_SCRIPTS = upd-updates run-anaconda anaconda-yum zramswapon zramswapoff zram-stats
+dist_scripts_SCRIPTS = upd-updates run-anaconda anaconda-yum zramswapon zramswapoff zram-stats anaconda-pre-log-gen
 dist_noinst_SCRIPTS  = upd-kernel makeupdates
 
-dist_bin_SCRIPTS = analog anaconda-cleanup instperf anaconda-disable-nm-ibft-plugin
+dist_bin_SCRIPTS = analog anaconda-cleanup instperf anaconda-disable-nm-ibft-plugin
```
Comment (bcl) [inline scripts/Makefile.am]: Unneeded space at the end of this line

**Comment by japokorn** [comment]
jenkins, test this please

**Comment by jkonecny12** [comment]
Looks good to me.

**Comment by japokorn** [comment]
Pushed.

**Pull request closed without merging**